src/main/drivers/gyro_sync.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * gyro_sync.c
+ *
+ *  Created on: 3 aug. 2015
+ *      Author: borisb
+ */
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "platform.h"
+#include "build_config.h"
+
+#include "common/axis.h"
+#include "common/maths.h"
+
+#include "drivers/sensor.h"
+#include "drivers/accgyro.h"
+#include "drivers/accgyro_mpu6050.h"
+#include "drivers/accgyro_spi_mpu6000.h"
+#include "drivers/gyro_sync.h"
+
+#include "sensors/sensors.h"
+#include "sensors/acceleration.h"
+
+#include "config/runtime_config.h"
+#include "config/config.h"
+
+extern gyro_t gyro;
+
+uint32_t targetLooptime;
+static uint8_t mpuDivider;
+
+bool getMpuDataStatus(gyro_t *gyro)
+{
+    bool mpuDataStatus;
+
+    gyro->intStatus(&mpuDataStatus);
+    return mpuDataStatus;
+}
+
+bool gyroSyncCheckUpdate(void) {
+    return getMpuDataStatus(&gyro);
+}
+
+void gyroUpdateSampleRate(uint32_t looptime, uint8_t lpf, uint8_t syncGyroToLoop) {
+    int gyroSamplePeriod;
+    int minLooptime;
+
+    if (syncGyroToLoop) {
+#ifdef STM32F303xC
+        if (lpf == INV_FILTER_256HZ_NOLPF2) {
+            gyroSamplePeriod = 125;
+
+            if(!sensors(SENSOR_ACC)) {
+                minLooptime = 500;   // Max refresh 2khz
+            } else {
+                minLooptime = 625;   // Max refresh 1,6khz
+            }
+        } else {
+            gyroSamplePeriod = 1000;
+            minLooptime = 1000;      // Full sampling
+        }
+#elif STM32F10X
+        if (lpf == INV_FILTER_256HZ_NOLPF2) {
+            gyroSamplePeriod = 125;
+
+            if(!sensors(SENSOR_ACC)) {
+                minLooptime = 625;   // Max refresh 1,6khz
+            } else {
+                minLooptime = 1625;  // Max refresh 615hz
+            }
+        } else {
+            gyroSamplePeriod = 1000;
+            if(!sensors(SENSOR_ACC)) {
+                minLooptime = 1000;  // Full sampling without ACC
+            } else {
+                minLooptime = 2000;
+            }
+        }
+#endif
+        looptime = constrain(looptime, minLooptime, 4000);
+        mpuDivider  = (looptime + gyroSamplePeriod -1 ) / gyroSamplePeriod - 1;
```

Comment:
This is a bit confusing - the -1 is mpu6xxx divider specific, it would be better to compute true value here 'number of gyro samples between loop invocation' and subtract 1 in MPU driver.
Or name the variable so that 'number of samples to discard before loop invocation' is obvious